src/components/LogRow/CollapsedRow/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { forwardRef } from "react";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { palette } from "@leafygreen-ui/palette";
+import { Overline } from "@leafygreen-ui/typography";
+import { ListRowProps } from "react-virtualized";
+import Icon from "components/Icon";
+import { size } from "constants/tokens";
+
+const { gray } = palette;
+
+interface CollapsedRowProps extends ListRowProps {
+  numCollapsed: number;
+}
+
+const CollapsedRow = forwardRef<any, CollapsedRowProps>((props, ref) => {
+  const { index, numCollapsed, ...rest } = props;
+
+  const lineText =
+    numCollapsed !== 1 ? `${numCollapsed} lines skipped` : "1 line skipped";
+
+  return (
+    <RowContainer {...rest} ref={ref} data-cy={`collapsed-row-${index}`}>
+      <Overline> {lineText} </Overline>
+      <StyledButton leftGlyph={<Icon glyph="Expand" />} size="xsmall">
+        All
+      </StyledButton>
+      <StyledButton leftGlyph={<Icon glyph="Expand" />} size="xsmall">
+        5 Above and Below
+      </StyledButton>
+    </RowContainer>
+  );
+});
+
+CollapsedRow.displayName = "CollapsedRow";
+
+const RowContainer = styled.div`
+  display: flex;
+  align-items: center;
+  overflow-y: hidden;
+
+  height: ${size.m} !important;
+  padding-left: ${size.s};
+  background-color: ${gray.light2};
+`;
```

Comment:
It looks like if there are more than a few rows this starts to break rendering. 
<img width="757" alt="image" src="https://user-images.githubusercontent.com/4605522/189758925-f377acb8-c2c0-4c9d-9b8f-e1a4b2aac055.png">

Reply:
I replicated this by implementing a rudimentary search functionality. 
Inside of LogWindow.tsx. i replaced `processedLogLines` with 
```
  const [search] = useQueryParam(QueryParams.Search, "");

  const processedLogLines = useMemo(() => {
    if (!search) {
      return [...Array(lineCount).keys()];
    }
    const regex = new RegExp(search, "i");
    const matchingSearches = [];
    for (let i = 0; i < lineCount; i++) {
      const line = getLine(i) || "";
      if (regex.test(line)) {
        matchingSearches.push(i);
      } else if (Array.isArray(matchingSearches[matchingSearches.length - 1])) {
        // @ts-expect-error
        matchingSearches[matchingSearches.length - 1].push(i);
      } else {
        matchingSearches.push([i]);
      }
    }
    return matchingSearches;
  }, [search, lineCount, getLine]);
  ```